### ISSUE TITLE
pytorch2onnx --verify failing for retinanet and output incorrect

### ISSUE THREAD

**Issue by pablovela5620** [open-mmlab/mmdetection#3882]
**Describe the bug**
Getting the following error from trying to convert `retinanet_r50_fpn_1x_coco.py` after running
```
python tools/pytorch2onnx.py configs/retinanet/retinanet_test_r50_fpn_1x_coco.py ~/0Dev/data/mmdetection/retinanet_r50_fpn_1x_coco_20200130-c2398f9e.pth --output-file ~/0Dev/data/mmdetection/retinanet_r50_fpn_1x_coco_20200130-c2398f9e.onnx --verify --input-img demo/demo.jpg

```

```
Traceback (most recent call last):
  File "tools/pytorch2onnx.py", line 170, in <module>
    normalize_cfg=normalize_cfg)
  File "tools/pytorch2onnx.py", line 85, in pytorch2onnx
    pytorch_result[0][:, 4], onnx_results[:, 4]
TypeError: list indices must be integers or slices, not tuple
```
It seems like the error is in the final assert statement. I tried changing

```python 
pytorch_result[0][:, 4], onnx_results[:, 4]
```
to
```python 
pytorch_result[0][0][:, 4], onnx_results[:, 4]
```
but then got the following error
```
ValueError: operands could not be broadcast together with shapes (9,) (6,) 
```
I tried visualizing the result since the conversion did technically work by adding the following to the end of the verify code
```python
# only compare a part of result
bbox_results = bbox2result(det_bboxes, det_labels, 80)
onnx_results = bbox_results[0]
mmcv.imshow_bboxes(input_img, bbox_results)
```
but got the following garbage output
![image](https://user-images.githubusercontent.com/25287427/95020090-fd251a80-0636-11eb-9ba7-037830269042.png)

**Environment**

```

sys.platform: linux
Python: 3.7.7 (default, May  7 2020, 21:25:33) [GCC 7.3.0]
CUDA available: True
GPU 0,1: GeForce GTX 1080 Ti
CUDA_HOME: None
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.6.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0
OpenCV: 4.4.0
MMCV: 1.1.4
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.2
MMDetection: 2.4.0+ced1c57

```

**Comment by ZwwWayne** [comment]
@RyanXLi Please have a check for that.

**Comment by RyanXLi** [comment]
Hi! The assertion you mentioned does have a bug and it has been fixed in [#3857](https://github.com/open-mmlab/mmdetection/pull/3857) which is waiting to be merged. For your information, the line should be changed to `pytorch_result[0][0][0][:4], onnx_results[0][:4]`.
Meanwhile, the model transformed to onnx is most likely to be correct. The image output seems wrong because the pytorch2onnx script resizes the image (to [800, 1216] by default) and the box results are based on this resized image. You will need to also resize the image when drawing bbox on it, and then it will seem visually correct.

**Comment by pablovela5620** [comment]
Perfect, modifying the code exactly how you said fixed everything. Here is the image now that I've resized the image correctly (500, 500) showing top 4. 
![image](https://user-images.githubusercontent.com/25287427/95084821-6adf4e00-06ec-11eb-8122-72f2129d20df.png)


So for a bit of clarity, the assertion statement only checks if the location of the bounding box is numerically close but NOT the confidence values? Also even though you only currently support retinanet officially, would this technically work with any single-stage detector?

Lastly the output of the onnx converted model is (100, 5) detection bbox and (100,) labels. bbox2results basically converts this into a list of numpy arrays for the number of classes (in this case its 1, but in the full case it would be 80 for coco?)
